sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -0,0 +1,232 @@
+from sympy import sin, cos, Matrix, sqrt, pi, expand_mul, Abs, acos, S
+from sympy.core.symbol import symbols
+from sympy.physics.mechanics import dynamicsymbols, Body, PinJoint
+from sympy.physics.mechanics.joint import Joint
+from sympy.physics.vector import Vector, ReferenceFrame
+from sympy.testing.pytest import raises
+
+t = dynamicsymbols._t
+
+
+def test_Joint():
+    parent = Body('parent')
+    child = Body('child')
+    raises(TypeError, lambda: Joint('J', parent, child))
+
+
+def test_pinjoint():
+    P = Body('P')
+    C = Body('C')
+    l, m = symbols('l m')
+    theta, omega = dynamicsymbols('theta_J, omega_J', real=True)
+
+    Pj = PinJoint('J', P, C )
+    assert Pj._name == 'J'
```

Comment:
Why isn't `Pj.name` used? This should be a public attribute.

Reply:
Should name be public? If a user changes name of joint after instantiating an object of that class then it may cause some problem with self generated dynamicsymbols as they are dependent on name.

Reply:
Make it public but immutable.

Reply:
There is a `@property` decorator for that.